enterprise/coderd/groups.go

```diff
@@ -102,36 +102,56 @@ func (api *API) patchGroup(rw http.ResponseWriter, r *http.Request) {
 	)
 	defer commitAudit()
 
-	currentMembers, currentMembersErr := api.Database.GetGroupMembers(ctx, group.ID)
-	if currentMembersErr != nil {
-		httpapi.InternalServerError(rw, currentMembersErr)
+	var req codersdk.PatchGroupRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
 		return
 	}
 
-	aReq.Old = group.Auditable(currentMembers)
+	// If the name matches the existing group name pretend we aren't
+	// updating the name at all.
+	if req.Name == group.Name {
+		req.Name = ""
+	}
 
-	var req codersdk.PatchGroupRequest
-	if !httpapi.Read(ctx, rw, r, &req) {
+	if group.ID == group.OrganizationID && req.Name != "" {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: fmt.Sprintf("Cannot rename the %q group!", database.AllUsersGroup),
+		})
 		return
 	}
 
-	if req.Name != "" && req.Name == database.AllUsersGroup {
+	if group.ID == group.OrganizationID && (req.DisplayName != nil && *req.DisplayName != "") {
```

Comment:
Both this if statement and the prior should use `group.IsEveryone()` instead of `g.ID == g.OrgID` for readability.